clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/RedisCacheConfig.groovy

```diff
@@ -43,13 +45,22 @@ class RedisCacheConfig {
   }
 
   @Bean
+  @ConfigurationProperties("caching.redis")
+  RedisCacheOptions.Builder redisCacheOptionsBuilder() {
+    return RedisCacheOptions.builder();
+  }
+
+  @Bean
+  RedisCacheOptions redisCacheOptions(RedisCacheOptions.Builder redisCacheOptionsBuilder) {
+    return redisCacheOptionsBuilder.build()
+  }
+
+  @Bean
   NamedCacheFactory cacheFactory(
     JedisSource jedisSource,
     ObjectMapper objectMapper,
-    @Value('${redis.maxMsetSize:250000}') int maxMsetSize,
-    @Value('${caching.maxMergeCount:2500}') int maxMergeCount,
-    @Value('${caching.hashing.enabled:true}') boolean enableHashing) {
-    new RedisNamedCacheFactory(jedisSource, objectMapper, maxMsetSize, maxMergeCount, enableHashing, null)
+    RedisCacheOptions redisCacheOptions) {
+    new RedisNamedCacheFactory(jedisSource, objectMapper, redisCacheOptions, null)
```

Comment:
this changes the config keys, I would roll in by updating the config to this so that it is valid for both the previous version and new version:
```
redis:
  maxMsetSize: ${caching.redis.multiOpSize} #todo - unused key after vXXX

caching:
  hashing:
    enabled: ${caching.redis.hashingEnabled} #todo - unused key after vXXX
  maxMergeCount: ${caching.redis.batchSize} #todo - unused key after vXXX
  redis:
    batchSize: 5000 #previous caching.maxMergeCount
    multiOpSize: 10000 #previous redis.maxMsetSize
    hashingEnabled: false #previous caching.hashing.enabled
```